### DISCUSSION TITLE
Embedding vectors differ between llama-server and llama-embedding

### DISCUSSION THREAD

**Discussion by louisjang** [ggml-org/llama.cpp#10885] [Q&A] [answered]
Hi,

While I'm playing with llama.cpp with text embedding model, I found that the embedding vectors differ between llama-server and llama-embedding.

First, I've encoded via llama-server as below.
```
$ llama-server  -m bge-small-en-v1.5/bge-small-en-v1.5-F16.gguf --embeddings -c 512
$ curl -X POST http://localhost:8080/embeddings -H "Content-Type: application/json"  -d '{
           "content": "Hello, how are you?"
         }
'
...
"embedding":[-0.035285793244838715,-0.054461050778627396,0.016111496835947037,-0.01932154968380928
...
```
I've encoded the same text with llama-embedding tool, and the result differ from the above one. 
```
$ llama-embedding -m bge-small-en-v1.5/bge-small-en-v1.5-F16.gguf -p 'Hello, how are you?' -c 512
embedding 0: -0.034889 -0.057268  0.046235 -0.039690 -0.019376 -0.024002  0.117010  0.032591 ...
```

I've tested several other models, but this issue persists.

If you have any hint or comment, please let me know.

Thanks,
Louis

**Comment by ggerganov** [comment, marked as answer]
Will be fixed in https://github.com/ggerganov/llama.cpp/pull/10872

**Reply by louisjang** [reply]
Thank you for letting me know. I have confirmed that the issue has been resolved.
